DOCSP-41978: Count documents (#108)
mongodb/docs-php-library: source/read/count.txt

a collection, and the ``MongoDB\Collection::estimatedDocumentCount()`` method returns
the estimated number of documents in a collection.

Sample Data
~~~~~~~~~~~

The examples in this guide use the ``companies`` collection in the ``sample_training``
database from the :atlas:`Atlas sample datasets </sample-data>`. To access this collection
from your PHP application, instantiate a ``MongoDB\Client`` that connects to an Atlas cluster
and assign the following values to your ``db`` and ``collection`` variables:

.. literalinclude:: /includes/read/count.php
    :language: php
    :dedent:
    :start-after: start-db-coll
    :end-before: end-db-coll

To learn how to create a free MongoDB Atlas cluster and load the sample datasets, see the
:atlas:`Get Started with Atlas </getting-started>` guide.

.. _php-accurate-count:

Retrieve an Accurate Count
--------------------------

Use the ``MongoDB\Collection::countDocuments()`` method to count the number of documents
in a collection. To count the number of documents that match specific search criteria,
pass a query filter to the ``countDocuments()`` method.

> To learn more about specifying a query, see the :ref:`php-specify-query` guide.

Author: Note: link broken until #103 is merged

       | **Type**: ``integer``

   * - ``maxTimeMS``
     - | The maximum amount of time in milliseconds that the operation can run.
       | **Type**: ``integer``

   * - ``skip``
     - | The number of documents to skip before counting documents.
       | **Type**: ``integer``

   * - ``readPreference``
     - | The read preference to use for the operation. To learn more, see
         :manual:`Read Preference </core/read-preference/>` in the Server manual.
       | **Type**: ``MongoDB\Driver\ReadPreference``

The following example uses the ``countDocuments()`` method to count the number of
documents in which the ``number_of_employees`` field has the value ``50`` and instructs the
operation to count a maximum of ``100`` results:

.. io-code-block::
   :copyable:

   .. input:: /includes/read/count.php
      :start-after: start-modify-accurate
      :end-before: end-modify-accurate
      :language: php
      :dedent:

   .. output::
      :visible: false

      Number of companies with 50 employees: 100

.. _php-estimated-count:

Retrieve an Estimated Count
---------------------------

You can retrieve an estimate of the number of documents in a collection by calling

> the ``estimatedDocumentCount()`` method. The method estimates the amount of

Reviewer: S: in other method references, you include the class that the method is part of

Customize Estimated Count Behavior
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

You can modify the behavior of the ``estimatedDocumentCount()`` method
by passing an array that specifies option values as a parameter. The
following table describes the options you can set in the array:

.. list-table::
   :widths: 30 70
   :header-rows: 1

   * - Option
     - Description

   * - ``comment``
     - | The comment to attach to the operation.
       | **Type**: any valid BSON type

   * - ``maxTimeMS``
     - | The maximum amount of time in milliseconds that the operation can run.
       | **Type**: ``integer``

   * - ``readConcern``
     - | The read concern to use for the operation. To learn more, see
         :manual:`Read Concern </reference/read-concern/>` in the Server manual.
       | **Type**: ``MongoDB\Driver\ReadConcern``

   * - ``readPreference``
     - | The read preference to use for the operation. To learn more, see
         :manual:`Read Preference </core/read-preference/>` in the Server manual.
       | **Type**: ``MongoDB\Driver\ReadPreference``

   * - ``session``
     - | The client session to associate with the operation.
       | **Type**: ``MongoDB\Driver\Session``

> The following example uses the ``estimatedDocumentCount()`` method to return an
> estimate of the number of documents in the collection and instructs the operation
> to run for a maximum of ``1000`` milliseconds:

Reviewer: ```suggestion
The following example uses the ``estimatedDocumentCount()`` method to return an
estimate of the number of documents in the collection and sets a timeout of
``1000`` milliseconds on the operation:
```
